### ISSUE TITLE
A notion of published and unpublished cards

### ISSUE THREAD

**Issue by jkomoros** [jkomoros/complexity-compendium#172]
Recently we added a feature where cards with zero-length body content are cosindered empty.

But there are some cases where there's provisional content (a stub) but the card is still effectively empty.

That implies that the "empty" boolean is one that is possible to toggle in edit mode, and it is simply auto set or unset if content is empty.

**Comment by jkomoros** [comment]
The question is if the value of no-content shoudl be PRECISELY Whether that empty boolean is true or not. If so, then newCard() shoudl return a card with isEmpty true, and we'll need a maintencne script to set that in production

**Comment by jkomoros** [comment]
it might be better to have a notion of 'unpublished' cards that are only visible to the authors or admins. When they are visible to those people they have a different background to differentiate them. When a new card is created it is marked as published:false. But when you save a card in the UI, if published isn't set to true and there's content, it will warn you and ask you to confirm before proceeding. 

Obviously there should be a filter for published/unpublished

Basically there's an extra filter implied in every collection that culls unpublished cards unless the user is the author or an admin. Users can also of course layer on published/unpublished filters to get rid of the part they don't want if they choose.

**Comment by jkomoros** [comment]
A lot of the stuff left in #171 as TODOs actually belongs here now if cards are unpublished

**Comment by jkomoros** [comment]
- [ ] Add a published/unpublished filter
- [ ] A maintenance task to set all existing cards to published: true
- [ ] A checkbox for published in the editor for cards
- [ ] Warn when saving if the card has content but is unpublished
- [ ] card-renderer renders cards that are unpublished as gray background color
- [ ] card-thumbnails for unpublished cards also get gray background-color
- [ ] card-links don't act as links to cards that are unpublished for non-authors; for authors they show as red
